var/spack/repos/builtin/packages/nccmp/package.py

```diff
@@ -0,0 +1,30 @@
+from spack import *
+import os
+
+class Nccmp(Package):
+    """Compare NetCDF Files"""
+    homepage = "http://nccmp.sourceforge.net/"
+    url      = "http://downloads.sourceforge.net/project/nccmp/nccmp-1.8.2.0.tar.gz"
+
+    version('1.8.2.0', '81e6286d4413825aec4327e61a28a580')
+
+    depends_on('netcdf')
+
+    def install(self, spec, prefix):
+        # Configure says: F90 and F90FLAGS are replaced by FC and
+        # FCFLAGS respectively in this configure, please unset
+        # F90/F90FLAGS and set FC/FCFLAGS instead and rerun configure
+        # again.
+        os.environ['FC'] = os.environ['F90']
+        del os.environ['F90']
+        try:
+            os.environ['FCFLAGS'] = os.environ['F90FLAGS']
+            del os.environ['F90FLAGS']
+        except KeyError:    # There are no flags
```

Comment:
The try/catch block is more Pythonic.  See the definition of __contains__:

      https://hg.python.org/cpython/file/2.7/Lib/_abcoll.py

```

    def __contains__(self, key):
<https://hg.python.org/cpython/file/2.7/Lib/_abcoll.py#l386>
try: <https://hg.python.org/cpython/file/2.7/Lib/_abcoll.py#l387>
      self[key]
<https://hg.python.org/cpython/file/2.7/Lib/_abcoll.py#l388>
except KeyError:
<https://hg.python.org/cpython/file/2.7/Lib/_abcoll.py#l389>
 return False <https://hg.python.org/cpython/file/2.7/Lib/_abcoll.py#l390>
       else: <https://hg.python.org/cpython/file/2.7/Lib/_abcoll.py#l391>
           return True

```

With that in mind, the way I coded it involves the fewest steps for the
interpreter to go through.

On Wed, Apr 27, 2016 at 10:48 AM, Ben Boeckel <notifications@github.com>
wrote:

> In var/spack/repos/builtin/packages/nccmp/package.py:
>
> > +
> > +    version('1.8.2.0', '81e6286d4413825aec4327e61a28a580')
> > +
> > +    depends_on('netcdf')
> > +
> > +    def install(self, spec, prefix):
> > +        # Configure says: F90 and F90FLAGS are replaced by FC and
> > +        # FCFLAGS respectively in this configure, please unset
> > +        # F90/F90FLAGS and set FC/FCFLAGS instead and rerun configure
> > +        # again.
> > +        os.environ['FC'] = os.environ['F90']
> > +        del os.environ['F90']
> > +        try:
> > +            os.environ['FCFLAGS'] = os.environ['F90FLAGS']
> > +            del os.environ['F90FLAGS']
> > +        except KeyError:    # There are no flags
>
> I'd prefer a if 'F90FLAGS' in os.environ than a try/except block.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/LLNL/spack/pull/848/files/1a585a6c748d9445ff691e766efba350259364b5#r61271503>
>